File: lab7 - całkowanie-numeryczne-1/prostokąty.py
def rectangular_rule(f, a, b, n):
    h = (b - a) / n
    integral = 0
    for i in range(n):
        x = a + i * h
        integral += f(x)
    integral *= h
    return integral

# Przykład użycia
f = lambda x: x**2
a = 0
b = 1
n = 100
result = rectangular_rule(f, a, b, n)
print(f"Przybliżona wartość całki metodą prostokątów: {result}")

File: lab7 - całkowanie-numeryczne-1/prostokąty.m
function integral = rectangular_rule(f, a, b, n)
    h = (b - a) / n;
    x = linspace(a, b-h, n);
    integral = h * sum(f(x));
end

% Przykład użycia
f = @(x) x.^2;
a = 0;
b = 1;
n = 100;
result = rectangular_rule(f, a, b, n);
disp(['Przybliżona wartość całki metodą prostokątów: ', num2str(result)])
